azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1422 @@
+        )
+        ext_loc_id = self.instance_record["extendedLocation"]["name"].lower()
+        ssc_spcs = list(
+            ssc_client.azure_key_vault_secret_provider_classes.list_by_resource_group(
+                resource_group_name=self.resource_group_name
+            )
+        )
+
+        ssc_spcs = [spc for spc in ssc_spcs if spc["extendedLocation"]["name"].lower() == ext_loc_id]
+        client_ids = [spc["properties"]["clientId"] for spc in ssc_spcs if "clientId" in spc["properties"]]
+        if client_ids:
+            self.instance_identities.extend([mid["id"] for mid in self.get_identities_by_client_id(client_ids)])
+
+        self._add_deployment(
+            key=StateResourceKey.SSC_SPC,
+            api_version=ssc_api_version,
+            data_iter=ssc_spcs,
+            depends_on=instance_resource_id_expr,
+            parameters=nested_params,
+        )
+
+        ssc_secretsyncs = list(
+            ssc_client.secret_syncs.list_by_resource_group(resource_group_name=self.resource_group_name)
+        )
+        ssc_secretsyncs = [
+            secretsync
+            for secretsync in ssc_secretsyncs
+            if secretsync["extendedLocation"]["name"].lower() == ext_loc_id
+        ]
+        if ssc_secretsyncs and ssc_spcs:
+            self._add_deployment(
+                key=StateResourceKey.SSC_SECRETSYNC,
+                api_version=ssc_api_version,
+                data_iter=ssc_secretsyncs,
+                depends_on=get_resource_id_by_parts(
+                    "Microsoft.Resources/deployments", self.active_deployment[StateResourceKey.SSC_SPC][-1]
+                ),
+                parameters=nested_params,
+            )
+
+    def _analyze_instance_identity(self):
+        target_instance = getattr(self.rcontainer_map[StateResourceKey.INSTANCE.value], "resource_state", {})
+        identity: dict = target_instance.get("identity", {}).get("userAssignedIdentities", {})
+
+        for rid in identity:
+            identity[rid] = {}
+            self.instance_identities.append(rid)
+
+    def add_deployment_by_key(self, key: StateResourceKey) -> Tuple[str, str]:
+        deployments_by_key = self.active_deployment.get(key, [])
+        symbolic_name = f"{key.value}s_{len(deployments_by_key)+1}"
+        deployment_name = f"concat(parameters('resourceSlug'), '_{symbolic_name}')"
+        deployments_by_key.append(deployment_name)
+        self.active_deployment[key] = deployments_by_key
+        return symbolic_name, deployment_name
+
+    def _add_deployment(
+        self,
+        key: StateResourceKey,
+        api_version: str,
+        data_iter: Iterable,
+        depends_on: Optional[Union[str, Iterable[str]]] = None,
+        parameters: Optional[dict] = None,
+        resource_group: Optional[str] = None,
+        subscription: Optional[str] = None,
+    ):
+        data_iter = list(data_iter)
+        if data_iter:
+            chunked_list_data = chunk_list(
```

Comment:
sorry stupid question, why do we chunk the deployment data?

Reply:
A deployment cannot deploy more than 800 resources. Also there is a deployment size limit.